### ISSUE TITLE
Unexpected SC2034: variable appears unused

### ISSUE THREAD

**Issue by alvinseville7cf** [koalaman/shellcheck#2149]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or "online"): 0.7.1
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

variable="Debug"
variable2=1

case "$variable2" in
  1)
    variable="a"
  ;;
  2)
    variable="b"
  ;;
esac
```

#### Here's what shellcheck currently says:
```
In main.sh line 11:
    variable="b"
    ^------^ SC2034: variable appears unused. Verify use (or export if used externally).

For more information:
  https://www.shellcheck.net/wiki/SC2034 -- variable appears unused. Verify u...
```

#### Here's what I wanted or expected to see:

No warnings expected.

**Comment by brother** [comment]
Assignment is not usage.
The warning appears at the last assignment as far as I can remember, not sure where I prefer the warning to appear.

**Comment by alvinseville7cf** [comment]
Thanks for your answer. Got it. :)
